packages/app/browser/src/app.ts

```diff
@@ -28,3 +28,14 @@ submit.addEventListener("click", () => {
 	document.cookie = `password=${password.value}`;
 	location.reload();
 });
+
+/**
+ * Notify user on load of page if previous password was unsuccessful
+ */
+const reg = new RegExp(`password=(\\w+);?`);
```

Comment:
Could you describe how this works for me? Like why `document.cookie.match`? How are you telling if the previous password was unsuccessful?

Reply:
Certainly!

This page is only ever loaded if a valid password cookie does not exist. You can see that on line 29 the original code calls `location.reload()` which would just load into the IDE because the key is set (assuming it's valid). 

Document cookies are weird. Rather than doing something logical like returning a map or object. The returned value from `document.cookie` is a single string value with all the cookies concatenated together and separated by semi-colons. So I'm using `string.prototype.match()`to apply a regex and search to see if a cookie named "password" exists in the cookie string.

So with the exception of the first load we can determine that if this page ever loads and "password" cookie exists than it must not be correct. As for the handling of first load, I'm using `document.referrer` (line 39) which returns null if there was/is no previous page load.

Reply:
Isn't it insecure to keep the password in the cookie like this?

Reply:
Oh nvm, not coder.com, I guess this is fine for self hosted.